src/models/hooks/activityReportObjectiveFile.js

```diff
@@ -0,0 +1,62 @@
+const { getSingularOrPluralData } = require('../helpers/hookMetadata');
+
+const recalculateOnAR = async (sequelize, instance, options) => {
+  // check to see if objectiveId or objectiveIds is validly defined
+  // when defined a more efficient search can be used
+  const objectiveIds = getSingularOrPluralData(options, 'objectiveId', 'objectiveIds');
+
```

Comment:
Again would be nice to have a comment for each path of the IF ELSE that explains what we do for each.